src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs

```diff
@@ -56,9 +56,12 @@ internal sealed class LanguageSettingsPersister : ForegroundThreadAffinitizedObj
             {
                 var languagePreferences = new LANGPREFERENCES3[1];
                 languagePreferences[0].guidLang = languageGuid;
-                Marshal.ThrowExceptionForHR(_textManager.GetUserPreferences4(pViewPrefs: null, pLangPrefs: languagePreferences, pColorPrefs: null));
 
-                RefreshLanguageSettings(languagePreferences);
+                // The function can potentially fail if that language service isn't installed
+                if (ErrorHandler.Succeeded(_textManager.GetUserPreferences4(pViewPrefs: null, pLangPrefs: languagePreferences, pColorPrefs: null)))
```

Comment:
+1 on getting this cherry picked sooner. I am currently hitting this exception when trying to run VSI tests on Willow install which doesn't have TS language service.